File: two-dimensional-array/twoDimensionalArray.js
const matrix = (array) => {

    const output = [...array].map((outerElement, outerIndex) => {
        return outerElement.map((innerElement, innerIndex) => {
            if(innerIndex === outerIndex && innerElement > -1) {
                return 1;
            };
            if(innerIndex === outerIndex && innerElement < 0) {
                return 0;
            };
            return innerElement;
        });
    });

    return output;
};

module.exports = { matrix };

//https://www.codewars.com/kata/581214d54624a8232100005f/train/javascript

File: python-venv/two_dimensional_array/kata_solution/two_dimensional_array.py
def matrix(array):

    output = []

    for i in range(len(array)):
        output.append([])

        for j in range(len(array[i])):
            if array[i] == array[j] and array[i][j] > -1:
                output[i].append(1)
            elif array[i] == array[j] and array[i][j] < 0:
                output[i].append(0)
            else:
                output[i].append(array[i][j])

    return output

#https://www.codewars.com/kata/581214d54624a8232100005f/train/python